### DISCUSSION TITLE
Problem Regarding to The Calculation of The Cost

### DISCUSSION THREAD

**Discussion by Fantasydog05** [DLR-VSDC/IEEE-MVC-2023#34] [Q&A] [answered]
Hello,
Whenever the battery SoC hits the lower limit, the simulation is forced to be stopped. We can observe that the corresponding cost is smaller than the cost of the baseline EMA which can finish the simulation. However, is it normal to have a smaller cost without completing the test track? Would you please explain if there are any penalties added for incomplete simulations?

**Comment by JBRDLR** [comment]
Dear @Fantasydog05 can you please provide some supplements such as plots or data files?

**Reply by Fantasydog05** [reply]
Our simulation stops at about 16XXX seconds because SoCb reaches the lower limit. Baseline EMA stops at 17XXX seconds. In this case, the cost is still lower than baseline, but the overall simulation time is less than the baseline. Is this acceptable?
![image](https://user-images.githubusercontent.com/87512543/218368169-f7294782-baf5-4231-9ced-3443ef1e8fa7.png)
![image](https://user-images.githubusercontent.com/87512543/218368200-ccd894b9-064d-4b48-b091-cfb73a73810e.png)

**Comment by JBRDLR** [comment]
@rpintodecastro @imeb can you help to answer the question it is not totally clear to me.

**Comment by imeb** [comment, marked as answer]
Dear @Fantasydog05 and @JBRDLR,
We have implemented an emergency stop as following under Performance Metrics subsystem. This would imply a safety on the battery usage. Competitors are invited to avoid reaching these extreme cases.
<img width="734" alt="EMG_stop" src="https://user-images.githubusercontent.com/77015600/218540284-35f8e31b-e204-4fed-9f69-03f1f5fb06d0.png">
<img width="806" alt="Safety_vals" src="https://user-images.githubusercontent.com/77015600/218542259-8bad04f1-03dd-4ba1-9980-dd59ad8b8025.png">
For the moment, this has not been yet included in our evaluations in the sense to penalize this scenario; but please keep in mind that if simulation is terminated due to this stopping enforcement, is not a valid solution. In future update of the repository, we might include this in the evaluation to penalize to a large number.
I hope I could address your doubt. If not, please let me know.
Best,
Iman
